pkg/chunk/delete_requests_store.go

```diff
@@ -0,0 +1,253 @@
+package chunk
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"flag"
+	"fmt"
+	"hash/fnv"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+type DeleteRequestStatus string
+
+const (
+	Received     DeleteRequestStatus = "0"
+	BuildingPlan DeleteRequestStatus = "1"
+	Deleting     DeleteRequestStatus = "2"
+	Processed    DeleteRequestStatus = "3"
+)
+
+var pendingDeleteRequestStatuses = []DeleteRequestStatus{Received, BuildingPlan, Deleting}
+
+// DeleteRequest holds all the details about a delete request
+type DeleteRequest struct {
+	RequestID string              `json:"request_id"`
+	UserID    string              `json:"-"`
+	StartTime model.Time          `json:"start_time"`
+	EndTime   model.Time          `json:"end_time"`
+	Selectors []string            `json:"selectors"`
+	Status    DeleteRequestStatus `json:"status"`
+	Matchers  [][]*labels.Matcher `json:"-"`
+	CreatedAt model.Time          `json:"created_at"`
+}
+
+// DeleteStore provides all the methods required to manage lifecycle of delete request and things related to it
+type DeleteStore struct {
```

Comment:
Naming is hard :smile: I had thought about it. I went with this name because it would have another table for storing cache gen numbers that would be introduced in another PR so it would not be just the delete requests that it stores but something related to it.